File: Stack/code.java
class Solution {
    public int solution(int[] A, int[] B) {
        // write your code in Java SE 8
        int N=A.length;
        Stack<Integer> upStream=new Stack<>();
        Stack<Integer> downStream=new Stack<>();
        for(int i=0;i<N;i++)
        {
            if(B[i]==0)
            {
              while(!downStream.empty() && downStream.peek()<=A[i])
              downStream.pop();
              if(downStream.empty())
              {
                  upStream.push(A[i]);
              }
              
            }
            else
            {
               downStream.push(A[i]);
            }
        }
        return upStream.size()+downStream.size();
    }
}

File: maxSilceSum/MaxProfit/code.js
function solution(A) {
    // write your code in JavaScript (Node.js 8.9.4)
    let N = A.length;
    let buy = A[0], sell = [0], maxProfit = 0;
    for (let stock of A) {
        if (stock > buy) {
            sell = stock;
        }
        else {

            buy = stock;
            sell = stock;
        }
        if (maxProfit < (sell - buy))
            maxProfit = sell - buy;
    }
    if (maxProfit < (sell - buy))
        maxProfit = sell - buy;
    return maxProfit;
}